### PR TITLE
Poly-commitment: more docs and comments

### PR DESCRIPTION
**Pull request by dannywillems**

### CODE REVIEW

File: poly-commitment/src/ipa.rs [hunk 1]
```diff
@@ -840,6 +898,13 @@ impl<G: CommitmentCurve> SRS<G> {
             .map(|(a, b)| *a * b)
             .fold(G::ScalarField::zero(), |acc, x| acc + x);
 
+        // Usually, the prover sends `combined_inner_product`` to the verifier
+        // So we should absorb `combined_inner_product``
+        // However it is more efficient in the recursion circuit
+        // to absorb a slightly modified version of it.
+        // As a reminder, in a recursive seeting, the challenges are given as a public input
```
Comment (dannywillems) [inline poly-commitment/src/ipa.rs:905]:
```suggestion
        // As a reminder, in a recursive setting, the challenges are given as a public input
```

**Review by marcbeunardeau88 — APPROVED** [review]

File: poly-commitment/src/ipa.rs [hunk 2]
```diff
@@ -140,15 +185,22 @@ pub fn combine_polys<G: CommitmentCurve, D: EvaluationDomain<G::ScalarField>>(
...
-    let mut plnm = plnm.to_dense_polynomial();
+    // Now, we will combine both evaluations and coefficients forms
+
+    // plnm will be our final combined polynomial. We first treat the
+    // polynomials in coefficients forms, which is simply scaling the
+    // coefficients and add them.
+    let mut plnm = plnm_coefficients.to_dense_polynomial();
+
     if !plnm_evals_part.is_empty() {
+        // n is the number of evaluations points, which is a multiple of the
```
Comment (dannywillems) [inline poly-commitment/src/ipa.rs:196]:
```suggestion
        // n is the number of evaluations, which is a multiple of the
```

File: poly-commitment/src/commitment.rs [hunk 3]
```diff
@@ -466,6 +476,7 @@ pub fn combined_inner_product<F: PrimeField>(
 }
 
 /// Contains the evaluation of a polynomial commitment at a set of points.
+#[derive(Clone)]
```
Comment (dannywillems) [inline poly-commitment/src/commitment.rs:479]: Is it still required?

Comment (marcbeunardeau88) [reply poly-commitment/src/commitment.rs:479]: nope

**Pull request merged**